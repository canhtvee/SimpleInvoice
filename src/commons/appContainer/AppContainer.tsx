import React from 'react';
import {
  Edge,
  NativeSafeAreaViewProps,
  SafeAreaView,
} from 'react-native-safe-area-context';
import {Colors} from '../../utils';

type EdgesType = 'lr' | 'lrt' | 'lrb' | 'lrtb';

export interface AppContainerProps
  extends Omit<NativeSafeAreaViewProps, 'edges'> {
  edges?: EdgesType;
}

const EdgesValues: Record<EdgesType, ReadonlyArray<Edge>> = {
  lr: ['left', 'right'],
  lrt: ['left', 'right', 'top'],
  lrb: ['left', 'right', 'bottom'],
  lrtb: ['left', 'right', 'top', 'bottom'],
};

export function AppContainer({
  children,
  style,
  edges = 'lr',
  ...safeViewProps
}: AppContainerProps) {
  return (
    <SafeAreaView
      edges={EdgesValues[edges]}
      style={[
        {
          flex: 1,
          justifyContent: 'flex-start',
          backgroundColor: Colors.background,
        },
        style,
      ]}
      {...safeViewProps}>
      {children}
    </SafeAreaView>
  );
}
