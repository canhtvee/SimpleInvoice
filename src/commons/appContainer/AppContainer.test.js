import React from 'react';
import {Text} from 'react-native';

import {cleanup} from '@testing-library/react-native';
import {create} from 'react-test-renderer';

import {AppContainer} from './AppContainer';

/**
 * Run test command
 * yarn test -u -t="AppContainer"
 */

afterEach(cleanup);

describe('AppContainer', () => {
  test('should render children', () => {
    const component = create(
      <AppContainer
        style={{backgroundColor: 'red'}}
        edges={['bottom', 'left', 'right']}>
        <Text>AppContainer_Test</Text>
      </AppContainer>,
    );

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.root.findAllByProps({type: 'Circle'}).length).toEqual(0);
    expect(component.root.findByType('Text').children).toEqual([
      'AppContainer_Test',
    ]);
  });
});
