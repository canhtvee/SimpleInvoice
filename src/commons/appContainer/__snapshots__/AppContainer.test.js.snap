// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer should render children 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "justifyContent": "flex-start",
      },
      {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Text>
    AppContainer_Test
  </Text>
</RNCSafeAreaView>
`;
